src/parser/jobs/nin/modules/Ninjutsu.js

```diff
@@ -8,71 +8,71 @@ import Module from 'parser/core/Module'
 import {TieredSuggestion, Suggestion, SEVERITY} from 'parser/core/modules/Suggestions'
 
 const DOTON_TICK_TARGET = 6
-const TCJ_DOTON_TICK_TARGET = 7
 const JUSTIFIABLE_DOTON_TICKS = 10
 
 export default class Ninjutsu extends Module {
 	static handle = 'ninjutsu'
 	static dependencies = [
 		'combatants',
-		'downtime',
 		'suggestions',
 	]
 
 	_hyotonCount = 0
 	_rabbitCount = 0
-	_dotonCasts = []
+	_dotonCasts = {
+		current: null,
+		history: [],
+	}
 
 	constructor(...args) {
 		super(...args)
-		this.addHook('cast', {by: 'player', abilityId: ACTIONS.HYOTON.id}, this._onHyotonCast)
-		this.addHook('cast', {by: 'player', abilityId: ACTIONS.RABBIT_MEDIUM.id}, this._onRabbitCast)
-		this.addHook('cast', {by: 'player', abilityId: ACTIONS.DOTON.id}, this._onDotonCast)
+
+		this.addHook('cast', {by: 'player', abilityId: [ACTIONS.HYOTON.id, ACTIONS.HYOTON_TCJ.id]}, () => { this._hyotonCount++ })
+		this.addHook('cast', {by: 'player', abilityId: ACTIONS.RABBIT_MEDIUM.id}, () => { this._rabbitCount++ })
+		this.addHook('cast', {by: 'player', abilityId: [ACTIONS.DOTON.id, ACTIONS.DOTON_TCJ.id]}, this._onDotonCast)
 		this.addHook('aoedamage', {by: 'player', abilityId: STATUSES.DOTON.id}, this._onDotonDamage)
+		this.addHook('removebuff', {by: 'player', abilityId: STATUSES.DOTON.id}, this._finishDotonWindow)
 		this.addHook('complete', this._onComplete)
```

Comment:
Could probably use a check that Katon or Gokka Mekkakyu actually hit multiple targets.

Reply:
I'm going to put that on the "do it in a future PR list". I'm more concerned with just correcting what already exists for this one.